common/achievement/util.ts

```diff
@@ -94,14 +94,20 @@ export async function getBucketContext(userID: string, relation?: string): Promi
 }
 
 export function transformPrismaJson(user: User, json: Prisma.JsonValue): AchievementContextType | null {
-    if (!json['match'] && !json['subcourse']) {
+    const keys = Object.keys(json);
+    if (!keys) {
         return null;
     }
```

Comment:
Do we even need this check?

Reply:
Yes, if we do not check for existing keys, the forEach loop on keys will throw an error, since forEach() is not a function for null / undefined.